webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -142,7 +155,29 @@ export class TasksComponent implements OnInit, OnDestroy {
       .then(taskDocs => {
         this.hasTasks = taskDocs.length > 0;
         this.loading = false;
-        this.tasksActions.setTasksList(this.hydrateEmissions(taskDocs));
+        this.hydrateEmissions(taskDocs)
+          .then((tasks) => {
+            // get lineages for all tasks
+            this.getLineagesFromTaskDocs(tasks)
+              .then((subjects) => {
+                const deepCopy = obj => JSON.parse(JSON.stringify(obj));
```

Comment:
What's the issue you're trying to resolve with a deepCopy? 

What is the structure of this `obj`?